cmd/plasmacli/subcmd/eth/prove.go

```diff
@@ -76,28 +69,51 @@ var proveCmd = &cobra.Command{
 
 // Returns transaction results for given position
 // Trusts connected full node
-func getProof(ctx context.CLIContext, addr ethcmn.Address, position plasma.Position) (*tm.ResultTx, []byte, error) {
-	// query for the output
-	key := append(addr.Bytes(), position.Bytes()...)
-	res, err := ctx.QueryStore(key, "utxo")
+func getProof(position plasma.Position) (*tm.ResultTx, []byte, error) {
+	ctx := context.NewCLIContext().WithTrustNode(true)
+
+	key := store.GetOutputKey(position)
+	hash, err := ctx.QueryStore(key, "outputs")
 	if err != nil {
 		return &tm.ResultTx{}, nil, err
 	}
 
-	utxo := store.UTXO{}
-	if err := rlp.DecodeBytes(res, &utxo); err != nil {
-		return &tm.ResultTx{}, nil, err
+	txKey := store.GetTxKey(hash)
+	txBytes, err := ctx.QueryStore(txKey, "outputs")
+
+	var tx store.Transaction
+	if err := rlp.DecodeBytes(txBytes, &tx); err != nil {
+		return &tm.ResultTx{}, nil, fmt.Errorf("Transaction decoding failed: %s", err.Error())
 	}
 
 	// query tm node for information about this tx
-	result, err := ctx.Client.Tx(utxo.MerkleHash, true)
+	result, err := ctx.Client.Tx(tx.Transaction.MerkleHash(), true)
 	if err != nil {
 		return &tm.ResultTx{}, nil, err
 	}
 
 	// Look for confirmation signatures
-	key = append([]byte("confirmSignature"), utxo.Position.Bytes()...)
-	sigs, err := ctx.QueryStore(key, "plasma")
+	// Ignore error if no confirm sig currently exists in store
+	var sigs []byte
+	if len(tx.SpenderTxs[position.OutputIndex]) > 0 {
+		queryPath := fmt.Sprintf("custom/utxo/tx/%s", tx.SpenderTxs[position.OutputIndex])
```

Comment:
Should we be reusing code from GetTx and such within the `subcmd/query` module.

I think we should move all the helper functions within the `subcmd/query` module to the `store` module or create a new top level `query` module.  Like `store.GetTx(ctx, hash)`

Reply:
having these generalized helpers defined within a top level package will make them easy to compose in the cli

Reply:
opened issue for this